### ISSUE TITLE
Standalone/LV2 not scaling up on 4k/HiDPI monitor (Linux)

### ISSUE THREAD

**Issue by zkf** [timothyschoen/PlugData#119]
See attached screenshot of standalone.

(When loaded in Ardour 6, the VST3 does scale correctly, while the LV2 does not.)

---

![image](https://user-images.githubusercontent.com/613216/169657443-7ab63387-31a6-49a3-b055-d5d81cd016c6.png)

**Comment by timothyschoen** [comment]
I noticed this as well, I think the best solution is to wait for JUCE 7, which has built-in LV2 support (and will also lead to a huge performance boost on some platforms)

**Comment by zkf** [comment]
Sounds good!

**Comment by dromer** [comment]
Just did a build of latest HEAD that uses JUCE7 and it works pretty well (some minor visual bugs on the standalone).

Maybe time to try this again? I think JUCE added some fixes for scaling issues in the mean time, so this 4K/HiDPI issue may be resolved (I don't have such a screen so can't test this).

**Comment by timothyschoen** [comment]
Nice, thanks for testing! I'll give it a try as well, the LV2 related issues should at least be fixed.

**Comment by zkf** [comment]
Unfortunately, for me the standalone build still appears tiny as in the screenshot. And I’m unable to load the LV2 plugin in Ardour 6. It reports (in the log window) `2022-06-25T17:10:51 [ERROR]: LV2: Failed to instantiate plugin https://github.com/timothyschoen/PlugData` and (in the console) `JUCE Assertion failure in juce_LV2_Client.cpp:1371`

**Comment by timothyschoen** [comment]
I noticed that the LV2 doesn't load anymore, I'm working on a fix. That assertion is helpful, I didn't even see that!

You have to make sure you actually set the resolution scaling factor to 2x, and not just set text/ui scaling to x2. I've tested it under Fedora and it works fine for me. Are you using Wayland or X11?

**Comment by dromer** [comment]
Current LV2 build on my system (debian stable) loads fine in Carla and Reaper btw.
Resizes nicely too (in Reaper the UI is embedded, works fine).

**Comment by zkf** [comment]
> You have to make sure you actually set the resolution scaling factor to 2x, and not just set text/ui scaling to x2. I've tested it under Fedora and it works fine for me. Are you using Wayland or X11?

I’m not sure if this is what you meant, but I’m on Manjaro and using KDE/Plasma on X11 and I’ve set the global scale to 200% in System Settings > Display and Monitor > Display Configuration.


The VST3 plugin running under Ardour is scaling properly though. And loading the LV2 plugin in Carla also scales correctly.

**Comment by timothyschoen** [comment]
Sounds like you have it set up correctly, so it's strange that the standalone won't scale... Also feels like an issue caused by JUCE.

The LV2 crashes are fixed for Reaper and Carla, but not yet for Ardour. This is actually because JUCE LV2 uses an LV2 feature that Ardour (6.1.2 at least) doesn't support, not much I can do about that :(

The VST3 also crashes for me when using Ardour (but not in Reaper), maybe I need to rebuild Ardour...

**Comment by timothyschoen** [comment]
I have fixed some issues with hi-dpi scaling that caused any dialog window (including the suggestions component and the settings) to be very small. Other than that, hi-dpi works for me on all platforms and hosts I've tried.

Linux: Reaper (VST3 and LV2), Carla (LV2), Ardour (VST3, JUCE LV2 plugins are currently unsupported so there's no way to know if it works)
Windows: Reaper (VST3), Ardour (VST3)
MacOS: Reaper (AU, VST3, LV2),  Logic Pro X (AU)

If the LV2 plugin doesn't scale up in Ardour (by the time they support all the necessary LV2 features), it would be outside of my control, so I'm going to close this issue.
